src/main/java/appeng/me/cache/NetworkMonitor.java

```diff
@@ -188,26 +188,17 @@ public boolean validForPass( final int i )
 	}
 
 	@Nullable
-	@SuppressWarnings( "unchecked" )
 	private IMEInventoryHandler<T> getHandler()
 	{
-		switch( this.myChannel )
-		{
-			case ITEMS:
-				return (IMEInventoryHandler<T>) this.myGridCache.getItemInventoryHandler();
-			case FLUIDS:
-				return (IMEInventoryHandler<T>) this.myGridCache.getFluidInventoryHandler();
-			default:
-		}
-		return null;
+		return this.myGridCache.getInventoryHandler( this.myChannel );
 	}
 
 	private Iterator<Entry<IMEMonitorHandlerReceiver<T>, Object>> getListeners()
 	{
 		return this.listeners.entrySet().iterator();
 	}
 
-	private T monitorDifference( final IAEStack original, final T leftOvers, final boolean extraction, final IActionSource src )
+	private T monitorDifference( final IAEStack<T> original, final T leftOvers, final boolean extraction, final IActionSource src )
 	{
 		final T diff = (T) original.copy();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "T". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1905)